Python/Product/Analysis/Analyzer/InterpreterScope.cs

```diff
@@ -289,15 +289,15 @@ internal bool RemoveVariable(string name, out VariableDef value)
         }
 
         internal virtual void ClearNodeValues() {
-            _nodeValues.Clear();
+            _nodeValues = new AnalysisDictionary<Node, NodeValue>();
```

Comment:
Yes. We only change them on the thread that accesses it, and we replace it specifically so that references obtained through the properties continue to point at the old dictionary.